boards/build.rs

```diff
@@ -24,6 +24,33 @@ fn main() {
         panic!("Boards must provide a `layout.ld` link script file");
     }
 
+    // The `RUSTFLAGS` that the Tock config files set can be easily overridden
+    // by command line flags. The build will still succeed but the resulting
+    // binary may be invalid as it was not built with the intended flags. This
+    // check seeks to prevent that. Our approach is we set a sentinel flag in
+    // our configuration file and then check that it is set here. If it isn't,
+    // the flags were overwritten and the build will be invalid.
+    //
+    // We only do this check if we are actually building for an embedded target
+    // (i.e., the TARGET is not the same as the HOST). This avoids a false
+    // positive when running tools like `cargo clippy`.
+    //
+    // If you are intentionally not using the standard Tock config files, set
+    // `cfg-tock-buildflagssentinel` in your cargo config to prevent this
+    // error.
```

Comment:
I totally disagree with this. This makes it impossible to override the flags passed to the compiler, which we may actually want!

The trick is just to avoid doing it by default, and know what's happening when you're doing it.

This is no different from, say, overriding `CC` to point to a different C-compiler, or setting `CFLAGS`.

Reply:
This is very different from overriding `CC`. This is like overriding `arm-none-eabi-gcc -std=gnu11 -Wbad-function-cast  -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition  -Wjump-misses-init  -frecord-gcc-switches -gdwarf-2 -Os -fdata-sections -ffunction-sections -fstack-usage -D_FORTIFY_SOURCE=2 -Wall -Wextra -I../.. -Wdate-time  -Wfloat-equal  -Wformat-nonliteral  -Wformat-security  -Wformat-y2k  -Winit-self  -Wmissing-declarations  -Wmissing-field-initializers  -Wmissing-format-attribute  -Wmissing-noreturn  -Wmultichar  -Wpointer-arith  -Wredundant-decls  -Wshadow  -Wunused-macros  -Wunused-parameter  -Wwrite-strings  -Wstack-usage=2048 -Wlogical-op  -Wtrampolines  -Wl,--emit-relocs -fPIC -mthumb -mfloat-abi=soft -msingle-pic-base -mpic-register=r9 -mno-pic-data-is-text-relative -isystem ../../lib/libtock-newlib-4.3.0.20230120/arm/arm-none-eabi/include -isystem ../../lib/libtock-libc++-13.2.0/arm/arm-none-eabi/include/c++/13.2.0 -isystem ../../lib/libtock-libc++-13.2.0/arm/arm-none-eabi/include/c++/13.2.0/arm-none-eabi -mcpu=cortex-m0 -march=armv6s-m -MF"build/cortex-m0/main.d" -MG -MM -MP -MT"build/cortex-m0/main.d@" -MT"build/cortex-m0/main.o" "main.c"` as one big hammer.

Reply:
Fine, it's like overriding `CFLAGS` (and wow :) ).

I agree it would be better if these were additive, rather than overriding each other (or at least if there was an option for them to be additive).

Reply:
https://github.com/rust-lang/cargo/issues/5376#issuecomment-1033180562 I think people normally expect a build system to concatenate when CFLAGS is set from the command line, rather than override

Reply:
e.g., C makefiles might contain `CFLAGS += -fPIC`, and then people can add additional CFLAGS on top of that from the environment variable if they wish.

I often use the trick of `CFLAGS=-Wno-error make` when I am building some old project and it is failing on some warning being treated as an error somewhere.